aws/internal/test/testing.go

```diff
@@ -0,0 +1,114 @@
+package test
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+const (
+	sentinelIndex = "*"
```

Comment:
Is it worth having the functions return an error if this sentinel value is not present or present more than once?

Reply:
I can return error if the last part in the address isn't `foo.bar.*`, since it always should